### ISSUE TITLE
FR: apply multiple "append after"-rules at once

### ISSUE THREAD

**Issue by wobbba** [mdelobelle/obsidian_supercharged_links#148]
This is the scenario:
I have source notes for papers I'm reading. I have properties of the papers that I'd like to display after the link, so I can sort and systematize them better. E.g. I want to display the topic with an emoji (🔄=Person-ReID, 🙂= face recognition, etc.) and I also want to mark which of the papers have code available (✔️=original code, 🔘= third party code, ❌= no code), among other things.
So say "Paper A" is about Person-ReID and has original code available, then I tag the source notes and want the link to display as 
`Paper A🔄✔️`

I see that it is implemented using the ::after pseudo-elements, so I get why this is not implemented. Maybe we could select a couple of selectors for which all combinations are generated? If we limit it to, say, 6 at a time then we would have 64 rules from those which is still manageable. 

But maybe that's not possible since the styling is done via the other plugin? Idk, anyway I'd love such a feature.

**Comment by HEmile** [comment]
Yes, i know what you mean and also have some elements where this would be useful. But indeed i wonder how to properly implement this. Creating combinations is indeed possible, but scales exponentially. Maybe a checkbox that allows this generation to be done only on selected rules might work.

**Comment by devinbaillie** [comment]
Just checking in and wondering whether this is on the roadmap?

A simple way to manage it would be to apply the rules in the order they are listed, and simply have a check box for each rule that says "stop here" or something like that, where if the user selects it, and that rule is applied, no further rules are checked.  If you don't select the option, then it continues checking additional rules until one is found that stops it (or until it finishes the list of rules).

Then any rules that are compatible (multiple prepends or postpends or different style types) would accumulate, and any rules that are contradictory (text colour or background colour) would simply overwrite the previously applied rule.  This would allow a lot more flexibility and control, without having to make rules for every possible combination of properties.

**Comment by HEmile** [comment]
Unfortunately, with the way Supercharged Links is implemented (ie, using CSS rules), this isn't as simple as you say it is. So given how little time I have to work on this project, it's not too likely I'll work on this soon. Feel free to PR though.

**Comment by devinbaillie** [comment]
Totally understand, and no worries.  And I didn't mean to imply it would be simple to implement, just that the interface could be simple.

**Comment by RomanMunar** [comment]
For those who only need two(1 for status, and type) icons to be side by side, just add a `float: left` to the appended icon.
```css
*[data-link-type]::after {
    float: left;
}
```
